ros2bag/ros2bag/verb/record.py

```diff
@@ -90,6 +90,13 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             '--qos-profile-overrides-path', type=FileType('r'),
             help='Path to a yaml file defining overrides of the QoS profile for specific topics.'
         )
+        parser.add_argument(
+            '--resilient-storage-writing', action='store_true',
+            help='Indicate preference for avoiding data corruption in case of crashes, '
+                 'at the cost of performance. Setting this flag disables optimization '
+                 'settings for storage (the defaut). When set, this flag will override '
+                 'corresponding settings in the config passed with --storage-config-file.'
```

Comment:
I think that this should not have top precedence, logically. It should modify the baseline settings, anything the user passes _explicitly_ (the `storage-config-file`) should always be respected.